### PR TITLE
Add new Nat Gateway resource

### PR DESCRIPTION
**Pull request by khannarheams**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 1]
```diff
@@ -0,0 +1,407 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2018-12-01"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Wrong API version. Same with examples.

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "200": {
+            "description": "Delete successful."
+          }
+        },
+        "x-ms-examples": {
+          "Delete nat gateway": { "$ref": "./examples/NatGatewayDelete.json" }
+        },
+        "x-ms-long-running-operation": true
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Please add `x-ms-long-running-operation-operations`
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R2010
For `DELETE` it should be `location`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-examples": {
+          "Create nat gateway": { "$ref": "./examples/NatGatewayCreate.json" }
+        },
+        "x-ms-long-running-operation": true
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Please add `x-ms-long-running-operation-operations`
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R2010
For `PUT` it should be `azure-async-operation`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json (hunk 1 above)
Comment (khannarheams) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Update all

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json (hunk 3 above)
Comment (khannarheams) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: thanks, updated

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json (hunk 2 above)
Comment (khannarheams) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: thanks, updated

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
Please, fix JSON errors https://dev.azure.com/azure-sdk/public/_build/results?buildId=15798

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+          },
+          "200": {
+            "description": "Delete successful."
+          }
+        },
+        "x-ms-examples": {
+          "Delete nat gateway": { "$ref": "./examples/NatGatewayDelete.json" }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation": {
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Should be `x-ms-long-running-operation-options`

**Review by anton-evseev — CHANGES_REQUESTED** [review]

**Comment by khannarheams** [comment]
fixed json errors

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+            "description": "Update successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update nat gateway tags": { "$ref": "./examples/NatGatewayUpdateTags.json" }
+        },
+        "x-ms-long-running-operation": true
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: Is this accurate? It looks like a synchronous operation based on responses and payload

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+      "description": "SKU of nat gateway"
+    },
+    "NatGatewayPropertiesFormat": {
+      "properties": {
+        "idleTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The idle timeout of the nat gateway."
+        },
+        "publicIpAddresses": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: please ensure linked access checks are defined for these referenced resourceids

**Review by KrisBash — CHANGES_REQUESTED** [review]
It looks like a spurious async operation = true flag is defined on the PATCH operation. Otherwise, looks good to me

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/azureFirewall.json [hunk 7]
```diff
@@ -736,7 +736,8 @@
          "description":"The action type of a rule collection",
          "enum":[
             "Allow",
-            "Deny"
+            "Deny", 
+            "Alert"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/azureFirewall.json]: Please revert this change. "Alert" value has been removed by AzureFirewall team https://github.com/Azure/azure-rest-api-specs/pull/5428

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+          },
+          "description": "An array of public ip prefixes associated with the nat gateway resource."
+        },        
+        "subnets": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./network.json#/definitions/SubResource"
+          },
+          "description": "Gets an array of references to the subnets using this nat gateway resource."
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]:
```suggestion
          "description": "An array of references to the subnets using this nat gateway resource."
```

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json (hunk 6 above)
Comment (khannarheams) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json]: thanks, this is updated here  https://msazure.visualstudio.com/_search?text=natgateway&type=code&lp=custom-Collection&result=DefaultCollection%2FOne%2FNetworking-nrp%2FGBmaster%2F%2Fsrc%2Fconfigs%2FEnvironmentConfigs%2Fcsm_registration_production.json&filters=ProjectFilters%7BOne%7DRepositoryFilters%7BNetworking-nrp%7D&_a=contents

**Review by KrisBash — APPROVED** [review]
LGTM thanks

**Comment by MikhailTryakhov** [comment]
@sergey-shandar can you please review

**Review by sergey-shandar — APPROVED** [review]
